packages/camera/camera_windows/windows/test/camera_plugin_test.cpp

```diff
@@ -142,26 +129,26 @@ TEST(CameraPlugin, CreateHandlerCallsInitCamera) {
 
   EXPECT_CALL(*camera_factory_, CreateCamera(MOCK_DEVICE_ID));
 
-  EXPECT_CALL(*result, ErrorInternal).Times(0);
-  EXPECT_CALL(*result, SuccessInternal(Pointee(EncodableValue(1))));
-
   CameraPlugin plugin(texture_registrar_.get(), messenger_.get(),
                       std::move(camera_factory_));
-  EncodableMap args = {
-      {EncodableValue("cameraName"), EncodableValue(MOCK_CAMERA_NAME)},
-      {EncodableValue("resolutionPreset"), EncodableValue(nullptr)},
-      {EncodableValue("enableAudio"), EncodableValue(true)},
-  };
-
-  plugin.HandleMethodCall(
-      flutter::MethodCall("create",
-                          std::make_unique<EncodableValue>(EncodableMap(args))),
-      std::move(result));
+
+  bool result_called = false;
+  std::function<void(ErrorOr<int64_t>)> create_result =
+      [&result_called](ErrorOr<int64_t> reply) {
+        EXPECT_FALSE(result_called);  // Ensure only one reply call.
+        result_called = true;
+        EXPECT_FALSE(reply.has_error());
+        EXPECT_EQ(reply.value(), 1);
+      };
+
+  plugin.Create(MOCK_CAMERA_NAME,
+                PlatformMediaSettings(PlatformResolutionPreset::max, true),
+                std::move(create_result));
+
+  EXPECT_TRUE(result_called);
```

Comment:
The test changes are extensive by lines, but pretty mechanical. The basic pattern to convert was:
- Replace each MockMethodResult with an inline lambda.
- Take `EXPECT_CALL` calls related to the `result` and convert them into assertions inside the lamba.
- Add the `result_called` boolean to allow asserting that the call actually happened, so that tests don't accidentally pass if a bug in the test means it just never calls the response at all.

The last step is a bit awkward, and makes things each test slightly longer, but on the other hand the use of a lambda instead of a mock will make doing more complicated assertions in tests much easier in the future. In practice almost all of the existing unit tests for success just assert that it succeeded, not that it had the right success value; fixing that was out of scope, but would now be easier to do (either as later backfill, or for new tests in the future).

Reply:
Thanks for the walkthrough. This makes sense. The end result is more flexible.